src/ray/core_worker/core_worker.cc

```diff
@@ -977,7 +978,8 @@ bool CoreWorker::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle,
             RAY_LOG(INFO) << "Owner's handle and creation ID " << object_id
                           << " has gone out of scope, sending message to actor "
                           << actor_id << " to do a clean exit.";
-            RAY_CHECK_OK(KillActor(actor_id, /*intentional=*/true));
+            RAY_CHECK_OK(
+                KillActor(actor_id, /*force_kill=*/true, /*no_reconstruction=*/false));
```

Comment:
@stephanie-wang Maybe the second argument should be `false`? Do you try to destroy the actor gracefully?